.github/workflows/rock-release-dev.yaml

```diff
@@ -0,0 +1,58 @@
+# The workflow canonical/observability/.github/workflows/rock-release-dev.yaml@main
+# https://github.com/canonical/observability/blob/34b7edb56094e8e3f01200ce49e01bf9dd6688ac/.github/workflows/rock-release-dev.yaml
+# does not allow setting a GHCR repo where to publish at the moment.
+# We thus copy it below and adapt it.
+
+name: Build ROCK and release dev tag to GHCR
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+
+jobs:
+  main:
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Checkout repository
+      uses: actions/checkout@v3
+
+    - name: Find the *latest* rockcraft.yaml
+      id: find-latest
+      run: |
+        latest_rockcraft_file=$(find $GITHUB_WORKSPACE -name "rockcraft.yaml" | sort -V | tail -n1)
+        rockcraft_dir=$(dirname ${latest_rockcraft_file#\./})
+        echo "latest-dir=$rockcraft_dir" >> $GITHUB_OUTPUT
+    - name: Build ROCK
+      uses: canonical/craft-actions/rockcraft-pack@main
+      with:
+        path: "${{ steps.find-latest.outputs.latest-dir }}"
+        verbosity: verbose
+      id: rockcraft
+
+    - name: Upload locally built ROCK artifact
+      uses: actions/upload-artifact@v3
```

Comment:
Artifacts are kept 90days by defaut. Please reduce that to the bare minimum (1week? 10days?).
Given that the artifact is pushed to GHCR, you could even clean it at the end of the workflow.